script/bump-version.js

```diff
@@ -0,0 +1,180 @@
+  const version = await nextVersion(opts.bump, currentVersion)
+  const components = semver.valid(semver.coerce(version)).split('.')
+
+  let pre
+  if (version.includes('-')) {
+    pre = `-${version.split('-')[1]}`
+    components[3] = version.split('-')[1].split('.')[1]
+  }
+
+  // print would-be new version and exit early
+  if (opts.dryRun) {
+    console.log(`new version number would be: ${version}\n`)
+    return 0
+  }
+
+  // update all version-related files
+  await Promise.all([
+    updateVersion(version),
+    updateInfoPlist(version),
+    updatePackageJSON(version),
+    tagVersion(version),
+    updateVersionH(components, pre),
+    updateWinRC(components)
+  ])
+
+  console.log(`Bumped to version: ${version}`)
+}
+
+// get next version for release based on [nightly, beta, stable]
+async function nextVersion (bumpType, version) {
+  if (utils.isNightly(version) || utils.isBeta(version)) {
+    switch (bumpType) {
+      case 'nightly':
+        version = await utils.nextNightly(version)
+        break
+      case 'beta':
+        version = await utils.nextBeta(version)
+        break
+      case 'stable':
+        version = semver.valid(semver.coerce(version))
+        break
+      default:
+        throw new Error('Invalid bump type.')
+    }
+  } else if (utils.isStable(version)) {
+    switch (bumpType) {
+      case 'nightly':
+        version = utils.nextNightly(version)
+        break
+      case 'beta':
+        throw new Error('Cannot bump to beta from stable.')
+      case 'stable':
+        version = semver.inc(version, 'patch')
+        break
+      default:
+        throw new Error('Invalid bump type.')
+    }
+  } else {
+    throw new Error(`Invalid current version: ${version}`)
+  }
+  return version
+}
+
+// update VERSION file with latest release info
+async function updateVersion (version) {
+  const versionPath = path.resolve(__dirname, '..', 'VERSION')
+  await writeFile(versionPath, version, 'utf8')
+}
+
+// update package metadata files with new version
+async function updatePackageJSON (version) {
+  ['package.json', 'package-lock.json'].forEach(async fileName => {
+    const filePath = path.resolve(__dirname, '..', fileName)
+    const file = require(filePath)
+    file.version = version
+    await writeFile(filePath, JSON.stringify(file, null, 2))
+  })
+}
+
+// update CFBundle version information and overwrite pre-existing file
+// TODO(codebytere): provide these version fields at GN build time
+async function updateInfoPlist (version) {
+  const filePath = path.resolve(__dirname, '..', 'atom', 'browser', 'resources', 'mac', 'Info.plist')
+  const xml = plist.parse(await readFile(filePath, { encoding: 'utf8' }))
+  const file = JSON.parse(JSON.stringify(xml))
+
+  file.CFBundleVersion = version
+  file.CFBundleShortVersionString = version
+
+  await writeFile(filePath, plist.build(file))
+}
+
+// push bump commit to release branch
+async function tagVersion (version) {
```

Comment:
What do you think about renaming this function, since it doesn't actually tag anything? Something like `commitVersionBump`.